src/arrayMethodSort.js

```diff
@@ -4,8 +4,23 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
+        if (
+          compareFunction
+            ? compareFunction(this[j], this[j + 1]) > 0
+            : String(this[j]) > String(this[j + 1])
+        ) {
+          const cash = this[j];
```

Comment:
The variable name 'cash' is a bit misleading as it usually refers to money. A more conventional name for a temporary variable in a swap operation is 'temp'.